mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -50,6 +50,8 @@ void LoopNestStateCollector::collect(Operation *opToWalk) {
       loadOpInsts.push_back(op);
     else if (isa<AffineWriteOpInterface>(op))
       storeOpInsts.push_back(op);
+    else if (!mlir::isMemoryEffectFree(op))
```

Comment:
If the op is not memoryEffectFree, LoopStateCollector should add this op to the list of MemoryEffectOps

Reply:
Is the above filter enough? Should it be  `else if (!mlir::isMemoryEffectFree(op) && !isa<mlir::memref::AllocOp, mlir::memref::DeallocOp>(op))` , instead of the above.

if only check `!mlir::ismemoryEffectFree(op)`, it conservatively includes alloc and dealloc ops as well. Because of this inclusion code like below fail to fuse successfully.

```module {
  func.func @testing2(%arg0: memref<10xf32>, %arg2 : memref <10xf32>) {
    %cst = arith.constant 1.000000e+00 : f32
    %cst_0 = arith.constant 2.000000e+00 : f32
    %1 = memref.alloc() : memref<12x12xf32>
    affine.for %arg1 = 0 to 10 { 
      %0 = affine.load %arg0[%arg1] : memref<10xf32>
    }
    affine.for %arg1 = 0 to 10 {
      %0 = affine.load %arg0[%arg1] : memref<10xf32>
      memref.dealloc %1  : memref<12x12xf32>
    }
    return
  }
} ```

Reply:
`memref.dealloc` and `memref.alloc` can be better handled in the MDG construction by tracking the memref they access similarly to load and store ops.

Reply:
Treating this as a separate issue for now and conservatively treating alloc and deallocate as having unknown side-effects.